src/serlo-editor/plugins/text/plugins/with-empty-lines-restriction.ts

```diff
@@ -0,0 +1,104 @@
+import { Editor, Element, Node, Path, Transforms } from 'slate'
+import { ReactEditor } from 'slate-react'
+
+// Slate normalization docs: https://docs.slatejs.org/concepts/11-normalizing
+// While editing a Text plugin: Two adjacent empty lines are allowed,
+// so that one of them could be transformed into another plugin,
+// using the plus button.
+// On Text plugin blur: Only one empty line is allowed.
+export const withEmptyLinesRestriction = (editor: Editor) => {
+  const { normalizeNode } = editor
+
+  editor.normalizeNode = (entry, { operation } = { operation: undefined }) => {
+    const [node, path] = entry
+    // custom operation type
+    const isBlur = (operation?.type as string) === 'blur_container'
+
+    if (shouldNodeBeRemoved(editor, node, path, isBlur)) {
+      Transforms.removeNodes(editor, { at: path })
+      // TODO: Show warning?
+      if (isBlur) ReactEditor.blur(editor)
+      return
+    }
+
+    normalizeNode(entry)
+  }
+
+  return editor
+}
+
+function isEmptyParagraph(editor: Editor, node: Node) {
+  return (
+    Element.isElement(node) && node.type === 'p' && Editor.isEmpty(editor, node)
+  )
+}
+
+function shouldNodeBeRemoved(
+  editor: Editor,
+  node: Node,
+  path: Path,
+  isBlur: boolean
+) {
+  // If the node is not an empty paragraph, it shouldn't be removed
+  if (!isEmptyParagraph(editor, node)) return false
+
+  // If the node has no previous sibling, it shouldn't be removed
+  const previousSiblingPath = Path.hasPrevious(path) && Path.previous(path)
+  if (!previousSiblingPath) return false
+  if (!Node.has(editor, previousSiblingPath)) return false
+
+  // On blur, every empty paragraph node with a previous sibling,
+  // that is also an empty paragraph, should be removed.
+  const previousSibling = Node.get(editor, previousSiblingPath)
+  const isPreviousSiblingEmptyParagraph = isEmptyParagraph(
+    editor,
+    previousSibling
+  )
+  if (isBlur) return isPreviousSiblingEmptyParagraph
+
+  // Collect the amount of empty paragraph siblings. It should never be more than 1.
+  let amountOfEmptyParagraphSiblings = isPreviousSiblingEmptyParagraph ? 1 : 0
```

Comment:
Maybe this could be simplified by only checking upwards: 
"If the current line has two empty lines above it -> Remove current line"

Reply:
I tried it but it lead to a different behavior and allowed three empty lines in some cases. Abandoned my idea because it works correctly right now.